### DISCUSSION TITLE
Forwarding the same value to two functions

### DISCUSSION THREAD

**Discussion by seeman1978** [nlohmann/json#3976] [General]
template<typename OutStringType = std::string, typename... Args>
inline OutStringType concat(Args && ... args)
{
    OutStringType str;
    str.reserve(concat_length(std::forward<Args>(args)...));
    concat_into(str, std::forward<Args>(args)...);
    return str;
}
The behavior of std::forward can be equivalent to the behavior of std::move.
It is not safe to forward a parameter more than once.

**Comment by gregmarr** [comment]
This was fixed in December.  https://github.com/nlohmann/json/pull/3889
